src/wallet/wallet.cpp

```diff
@@ -3930,6 +3938,7 @@ CWallet* CWallet::CreateWalletFromFile(const std::string& name, const fs::path&
     int64_t nStart = GetTimeMillis();
     bool fFirstRun = true;
     CWallet *walletInstance = new CWallet(name, CWalletDBWrapper::Create(path));
+    LOCK(walletInstance->cs_wallet);
     DBErrors nLoadWalletRet = walletInstance->LoadWallet(fFirstRun);
```

Comment:
Note that `LoadWallet` takes `LOCK2(cs_main, cs_wallet)`, so taking `cs_wallet` before `cs_main` leads to a deadlock in case another thread takes `cs_main` and then `cs_wallet`, no?

Reply:
Thanks. Now fixed. See https://github.com/bitcoin/bitcoin/pull/11634#discussion_r173839102.